spi/src/main/java/org/opensearch/security/spi/resources/ResourceProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.security.spi.resources;
+
+public class ResourceProvider {
+    private final String resourceType;
+    private final String resourceIndexName;
+    private final ResourceParser resourceParser;
+
+    public ResourceParser getResourceParser() {
+        return resourceParser;
+    }
+
+    public String getResourceIndexName() {
```

Comment:
If I see it correctly, `getResourceIndexName()` is not used anywhere. What would it be used for?